tests/integration/security/testdata/authz/tcp-L7.yaml.tmpl

```diff
@@ -0,0 +1,17 @@
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: {{ .To.ServiceName }}
+spec:
+  selector:
+    matchLabels:
+      "app": "{{ .To.ServiceName }}"
+  action: DENY
+  rules:
+    - to:
+        - operation: # HTTP
+            methods: [ "GET" ]
+    - to:
+        - operation: # HTTP and TCP
+            methods: [ "GET" ]
+            ports: [ "{{ (.To.PortForName `tcp`).WorkloadPort }}" ]
```

Comment:
This policy contains "ports". Is it a L7 only policy?

Reply:
So we check the policy rule by rule. In case of deny, any one rule matches then the request will be denied. In this case one rule has pure L7 attributes and other one has mixed (L7 and L4). So any request to any port will always be matched with rule1 (pure L7), hence it denies the request. Let's call this kind of policy as L7 policy.

Reply:
So this test case TestAuthz_TCP_L7 is similar to the mixed test case TestAuthz_TCP_Mixed as both of them have L4 and L7 attributes in the authz policy. Maybe the test names can be TestAuthz_TCP_Mixed_1 TestAuthz_TCP_Mixed_2.